compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/BodyResolveContext.kt

```diff
@@ -486,6 +486,11 @@ class BodyResolveContext(
         return withTowerDataCleanup {
             addLocalScope(FirLocalScope())
             if (function is FirSimpleFunction) {
+                // Make all value parameters available in the local scope so that even one parameter that refers to another parameter,
+                // which may not be initialized yet, could be resolved. We will detect and report such case in checkers.
```

Comment:
Suggestion: Be specific and say "in FirFunctionParameterChecker" instead of "in checkers" (also below in `forConstructorParameters`)